### ISSUE TITLE
Links to 'fragment' position within a notebook scroll out of view when cells render

### ISSUE THREAD

**Issue by jrus** [observablehq/feedback#42]
**Context: ** It is often convenient to link directly to a section of a long web document, and share that link outside the context of the page. The traditional method on the web is to use a 'fragment' link using a `#`. Observable attaches an `id` attribute with the cell name to the associated `div` in the notebook chrome, which makes fragment links from within the page scroll to the right place. Unfortunately when an Observable notebook initially loads, the vertical positions of all of these `div`s are compacted together near the top of the page, so that is where the browser scrolls, but after a small delay when notebook cells start rendering, their height increases dramatically, pushing the linked-to cell far down below the browser scroll position, typically stranding the reader at an arbitrary place near the top of the page with no easy way to get down to the desired section of the document.

**Steps to Reproduce:** Open a new browser tab pointed at a notebook link with a fragment identified, for example https://observablehq.com/@jrus/scpie#Contours

**Expected behavior:** The page should be scrolled to the section entitled "Sector boundary contours”.

**Actual behavior:** The page ends up scrolled part-way down the section entitled "Introduction"

Behavior is roughly the same in Mac Safari, Chrome, and Firefox.

**Proposed solution:* The platform could cache an expected height for each notebook cell (ideally cache a couple variants based on the most common notebook widths, e.g. one for a full-width desktop window and one for a typical smartphone) by pre-rendering the notebook offline and measuring the cell heights, and then using those as dummy values until the notebook fully renders. This should most of the time leave the scroll position in at least approximately the correct position before/after cell rendering.

**Comment by tomlarkworthy** [comment]
yeah a cold click on a link to cell URL does not work in the majority of cases, and it would be very useful for knowledge sharing if it did (particularly for deeper notebooks)
